cmake/winml_unittests.cmake

```diff
@@ -192,7 +192,7 @@ add_winml_test(
   SOURCES ${winml_test_api_src} ${winml_test_api_redist_only_src}
   LIBS winml_test_common
```

Comment:
I think we need to define "BUILD_INBOX" for winml_test_common

Reply:
function(add_winml_test) does this:
  if (onnxruntime_WINML_NAMESPACE_OVERRIDE STREQUAL "Windows")
    target_compile_definitions(${_UT_TARGET} PRIVATE "BUILD_INBOX=1")
  endif()

Reply:
does winml_test_common depend on the flag?

Reply:
spoke offline, winrt_headers.h is used in common.

While it doesnt break the build with the headers included, it is good to prevent their use as they are defunct on Windows builds.

Added BUILD_INBOX=1 check